tests/test_ml.py

```diff
@@ -71,12 +47,22 @@ def test_uploading_onnx_model(self):
                     description="A simple decision tree model",
                 )
             )
+            self.assertEqual(str(res_name), model_name)
+
+            # Now test permission setting
```

Comment:
what exactly do you test here? shouldn't you try to access the datasets with another user before and after adding the permission?